### PR TITLE
SK-267 added support for outdated session restore

### PR DESCRIPTION
**Pull request by Oleksandr1414**
* added `message_decryption_failed` handler
* fixed local storage for messages

### CODE REVIEW

File: src/services/conversationsService.js [hunk 1]
```diff
@@ -294,6 +295,14 @@ class ConversationsService {
   async getParticipantsByIds(params) {
     return await api.getUsersByIds(params);
   }
+
+  findOpponentIdForPrivateConversationByCid(cid, currentUserId) {
```
Comment (IgorKhomenko) [inline src/services/conversationsService.js:299]: it seems we already have this in `src/utils/user/get_opponent_id.js`

File: src/store/indexedDB.js [hunk 2]
```diff
@@ -16,6 +16,12 @@ class IndexedDB {
     );
   }
 
+  markDecryptionFailedMessages(mids) {
```
Comment (IgorKhomenko) [inline src/store/indexedDB.js:19]: So we have 2 functions now to update statuses:

```
  markMessagesAsRead(mids) {
    this.db.messages.bulkUpdate(
      mids.map((id) => ({ key: id, changes: { status: "read" } }))
    );
  }

  markDecryptionFailedMessages(mids) {
    this.db.messages.bulkUpdate(
      mids.map((id) => ({ key: id, changes: { status: "decryption_failed" } }))
    );
  }
```

I propose to universalize it:

```
  updateMessagesStatus(mids, status) {
    this.db.messages.bulkUpdate(
      mids.map((id) => ({ key: id, changes: { status } }))
    );
  }
```

**Review by IgorKhomenko — APPROVED** [review]
approved with minor comments

File: src/store/indexedDB.js (hunk 2 above)
Comment (Oleksandr1414) [reply src/store/indexedDB.js]: I had the same idea), but I decided that every time the method is called, writing the status would look not good

Also fixed this comment for Messages redux

**Pull request merged**